attrs.go

```diff
@@ -56,6 +56,50 @@ type FileStat struct {
 	Extended []StatExtended
 }
 
+func (fs FileStat) MarshalTo(flags FileAttrFlags) []byte {
```

Comment:
Either the signature should be `Marshal(FlagAttrFlags) []byte` or `MarshalTo([]byte, FlagAttrFlags) []byte`.

I recommended the latter, because the `marshalFileInfo` function was going to need to marshal the `FlagAttrFlags` first, and then call this function. Using an `b = append(b, stat.Marshal(flags)...)` would have to do at least two allocations, the first to hold the buffer marshaled by `stat.Marshal()` returned to `marshalFileInfo`, then the second, in `marshalFileInfo` to grow the buffer to copy in the data from the `stat.Marshal` call.

Sorry about being so nitpicky and pedantic on these matters, but we’ve had to do a lot of profiled optimization that demonstrates that these allocations and copies can significantly impact the execution time, and there’s a whole ton of learnings about how to properly juggle memory to avoid unnecessary allocations and copies from implementing the `internal/ssh/filexfer/` package.

Reply:
I must admit that I was puzzled when I saw that internal/ssh/filexfer was only used by the ls code as far as I could tell.  But good point on performance, I'll see about that tweak.

Reply:
It’s sitting there with the intent of prototyping v2 work, and eventually rolling out as v2. I did a big refactor, but then other PRs got in the way, and it diverged from the main branch enough to eventually need to be abandoned. 🙁